pkg/distro/image_config.go

```diff
@@ -94,6 +94,9 @@ type ImageConfig struct {
 	LockRootUser *bool
 
 	IgnitionPlatform *string
+
+	// How to handle the /etc/machine-id file
+	FirstBootStrategy *osbuild.MachineIdFirstBoot
```

Comment:
In general, we want to avoid using `osbuild` stage options or types as types in `ImageConfig`.

I'm also not too keen on the name. `FirstBootStrategy` sounds way too general, while all it does is manage `machine-id`... :thinking:

Reply:
`MachineIdStrategy`?

Reply:
`MachineIdFirstBoot`? :roll_eyes:  I like your suggestion, but the values make it weird.

I'll let others comment since there isn't a good option.

Reply:
> Yes, but the status quo right now is to use them. Do I need to introduce another type and then directly convert/map that onto osbuild types?

It is, but we want to move away from it. So, it seems natural not to use it for newly added options. So yes, ideally, you should add a new type to https://github.com/osbuild/images/tree/main/pkg/customizations if it can't be represented by a primitive type.

On the other hand, we do not need to necessarily map the stage options 1:1 to the `ImageConfig` option. Maybe adding just something like `MachineIdUninitialized bool` would be good enough? And the `OS` pipeline would just set the correct stage option if `true`.

Reply:
> On the other hand, we do not need to necessarily map the stage options 1:1 to the `ImageConfig` option. Maybe adding just something like `MachineIdUninitialized bool` would be good enough? And the `OS` pipeline would just set the correct stage option if `true`.

This should cover everything we're doing with the stage for now and we can expand on it later if we need to.